sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala

```diff
@@ -70,8 +72,16 @@ case class HadoopFsRelation(
   }
 
   override def sizeInBytes: Long = {
+    val isCatalogFileIndex = location.isInstanceOf[CatalogFileIndex]
     val compressionFactor = sqlContext.conf.fileCompressionFactor
-    (location.sizeInBytes * compressionFactor).toLong
+    val defaultSize = (location.sizeInBytes * compressionFactor).toLong
+
```

Comment:
```
location match {
  case cfi: CatalogFileIndex if sparkSession.sessionState.conf.fallBackToHdfsForStatsEnabled =>
    DDLUtils...
  case _ => defaultSize
}
```
maybe?
Does it make sense to push the check for `sparkSession.sessionState.conf.fallBackToHdfsForStatsEnabled` into the method?

Reply:
Thanks. Updated. 

> Does it make sense to push the check for sparkSession.sessionState.conf.fallBackToHdfsForStatsEnabled into the method?

The method `sizeInBytesFallBackToHdfs` is supposed for getting sizeInBytes from hdfs, if user enable the `fallback` configuration. I am not sure about moving configuration check to the method `sizeInBytesFallBackToHdfs`.